Incorrect no-use-before-define

**What version of TypeScript are you using?**
3.1.3

**What version of `typescript-eslint-parser` are you using?**
21.0.0

**What code were you trying to parse?**

```ts
function test(file: Blob) {
  const slice: typeof file.slice =
    file.slice || (file as any).webkitSlice || (file as any).mozSlice
  return slice
}
```

**What did you expect to happen?**
No error should be in `eslint` output

**What happened?**
```
  41:28  error  'slice' was used before it was defined  no-use-before-define
```

This worked fine with `typescript-eslint-parser@20.0.0`. `file.slice` is defined, as `file` is a `Blob` argument. I think the parser mistakes it with the `const slice`.

--- a/analyze-scope.js
+++ b/analyze-scope.js
@@ -385,6 +385,15 @@ class Referencer extends OriginalReferencer {
     }
 
     /**
+     * Create reference objects for the object part. (This is `obj.prop`)
+     * @param {TSTypeQuery} node The TSTypeQuery node to visit.
+     * @returns {void}
+     */
+    TSQualifiedName(node) {
+        this.visit(node.left);
+    }
+
+    /**
      * Create reference objects for the references in computed keys.
      * @param {TSPropertySignature} node The TSPropertySignature node to visit.
      * @returns {void}

--- a/tests/lib/scope-analysis.js
+++ b/tests/lib/scope-analysis.js
@@ -351,4 +351,23 @@ export default class ListModalStore {
 
         expect(messages).toStrictEqual([]);
     });
+
+    test("https://github.com/eslint/typescript-eslint-parser/issues/550", () => {
+        const code = `
+function test(file: Blob) {
+  const slice: typeof file.slice =
+    file.slice || (file as any).webkitSlice || (file as any).mozSlice
+  return slice
+}
+`;
+        const config = {
+            parser: "typescript-eslint-parser",
+            rules: {
+                "no-use-before-define": "error"
+            }
+        };
+        const messages = linter.verify(code, config, { filename: "issue.ts" });
+
+        expect(messages).toStrictEqual([]);
+    });
 });

